toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1422,35 +1422,37 @@ func Chage(installChroot *safechroot.Chroot, passwordExpirationInDays int64, use
 	return fmt.Errorf(`user "%s" not found when trying to change the password expiration date`, username)
 }
 
-func ConfigureUserGroupMembership(installChroot *safechroot.Chroot, username string, primaryGroup string,
-	secondaryGroups []string,
-) (err error) {
+func ConfigureUserGroupMembership(installChroot *safechroot.Chroot, username string, primaryGroup string, secondaryGroups []string) (err error) {
+	if installChroot == nil {
```

Comment:
You should probably just make `updateUserGroupMembershipCore` public and let callers use that one if they want to run without providing a chroot context.

Reply:
Decided to keep it as is in the new pr, as we are making sshkeys as a param in the original function